01-cloudformation/cloudformation1.2.2.yaml

```diff
@@ -0,0 +1,67 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: My most minimal template possible
+
+Parameters:
+    BucketParameter:
+        Description: BucketParameter
+        Type: String
+    UserParameter:
+        Description: UserParameter
+        Type: String
+
+Conditions:
+  PreferredRegion: !Equals 
+    - !Ref AWS::Region
+    - us-east-1
+
+Resources:
+  Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName:
+        'Fn::If':
+          - PreferredRegion
+          - !Join
```

Comment:
Try using the !Sub function to write/manage less code. https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-sub.html